assets/js/modules/analytics/datastore/settings.test.js

```diff
@@ -656,6 +767,79 @@ describe( 'modules/analytics settings', () => {
 				).toThrow( INVARIANT_INVALID_PROFILE_SELECTION );
 			} );
 
+			it( 'does not require a valid propertyID when the `ga4Reporting` feature flag is enabled', () => {
+				enabledFeatures.add( 'ga4Reporting' );
+
+				registry.dispatch( MODULES_ANALYTICS ).setSettings( {
+					...validSettings,
+					accountID: '12345',
+					propertyID: null,
+				} );
+
+				registry
+					.dispatch( MODULES_ANALYTICS )
+					.receiveGetExistingTag( tagWithPermission.propertyID );
+
+				expect(
+					registry.select( MODULES_ANALYTICS ).canSubmitChanges()
+				).toBe( true );
+
+				expect( () =>
+					registry
+						.select( MODULES_ANALYTICS )
+						.__dangerousCanSubmitChanges()
+				).not.toThrow( INVARIANT_INVALID_PROPERTY_SELECTION );
+			} );
+
+			it( 'does not require a valid profileID when the `ga4Reporting` feature flag is enabled', () => {
+				enabledFeatures.add( 'ga4Reporting' );
+
+				registry.dispatch( MODULES_ANALYTICS ).setSettings( {
+					...validSettings,
+					accountID: '12345',
+					profileID: null,
+				} );
+
+				registry
+					.dispatch( MODULES_ANALYTICS )
+					.receiveGetExistingTag( tagWithPermission.propertyID );
+
+				expect(
+					registry.select( MODULES_ANALYTICS ).canSubmitChanges()
+				).toBe( true );
+
+				expect( () =>
+					registry
+						.select( MODULES_ANALYTICS )
+						.__dangerousCanSubmitChanges()
+				).not.toThrow( INVARIANT_INVALID_PROFILE_SELECTION );
+			} );
+
+			it( 'does not require a valid internalWebPropertyID when the `ga4Reporting` feature flag is enabled', () => {
+				enabledFeatures.add( 'ga4Reporting' );
+
+				registry.dispatch( MODULES_ANALYTICS ).setSettings( {
+					...validSettings,
+					accountID: '12345',
+					propertyID: null,
```

Comment:
Shouldn't `propertyID` be omitted here to retain its valid value from `validSettings`?